src/libcore/ptr.rs

```diff
@@ -425,6 +746,11 @@ pub unsafe fn write_unaligned<T>(dst: *mut T, src: T) {
 /// to not be elided or reordered by the compiler across other volatile
 /// operations.
 ///
+/// Memory read with `read_volatile` should almost always be written to using
+/// [`write_volatile`].
```

Comment:
I am kind of left wondering when isn't this the case or a link to a place where one could read more about this.

Reply:
Well I wouldn't know...

Would you prefer me to remove that remark?

Reply:
@rkruppe , @japaric do you know ? Otherwise we can just open an issue about this to see if anyone has any good examples and just link to it. 

I've only used volatile reads and writes for interfacing with sensors and controls, but the addresses i could read from where not the same addresses as the ones I could write to so... i've never done volatile reads and writes to the same address :/ I could imagine there are devices that use the same address for I/O, so you have to do both volatile reads and volatile writes to the same address, but those operations are kind of independent, as in I don't think that anybody expect a volatile read to return the same value that was written with a previous volatile write. 

I have no idea what else people use volatile for, or what volatile reads/write do with "real" memory, multithreading, ... I thought one should use atomics for that.

Reply:
> what happens with them with concurrenct accesses (i've always used atomics for that), etc. :/

As you should, volatile is not for concurrency. ;) (The docs even say that by now. :D )

Reply:
I believe there are MMIO regions that are both read and written. Another use for volatile r/w accesses are values that need to be shared with signal handlers. I am not aware of any uses for mixing volatile and non-volatile accesses, usually that's plain incorrect. I would be comfortable just removing the "almost always", i.e., saying something like

> Memory accessed with `read_volatile` or `write_volatile` should not be accessed with non-volatile operations.

This wouldn't declare it to be UB, but would get the point across without raising the question of "when not?"

Reply:
Done, thanks!